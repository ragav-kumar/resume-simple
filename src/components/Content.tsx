import React from 'react'
import { Route, Switch } from 'react-router-dom';
import { sections } from '../utils/db';
import { Competency, Contact, Education, History, Profile } from "../pages";
import styled from "styled-components";
import { routes } from "../utils/types";

export const Content = () => {
	const menuList = sections.slice(1).map<string>(( { path } ) => path);
	// For the menu, Profile goes on top. In here, it needs to be at the bottom because of react-router
	// PDF and contact are not defined in sections because of reasons.
	const notInList = [
		routes.contact,
		routes.pdf,
		sections[0].path,
	];
	return (
		<Wrap>
			<Switch>
				{[ ...menuList, ...notInList ].map<JSX.Element>(( path, index ) => (
					<Route key={index} path={path}>
						{mapContent(path)}
					</Route>
				))}
			</Switch>
		</Wrap>
	);
}

const mapContent = ( path: string ): JSX.Element => {

	// noinspection FallThroughInSwitchStatementJS
	switch ( path ) {
		case routes.competency:
			return <Competency/>;
		case routes.education:
			return <Education/>;
		case routes.history:
			return <History/>;
		case routes.contact:
			return <Contact/>;
		default:
			return <Profile/>;
	}
}
const Wrap = styled.div`
	grid-area: content;
	height: 100%;
	max-width: 1024px;
	align-self: flex-start;
	overflow-y: auto;
	margin: 0 1vw;
`;
