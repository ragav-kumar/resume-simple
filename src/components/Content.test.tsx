import React from "react";
import renderer from "react-test-renderer";
import { Content } from "./Content";
import { BrowserRouter } from "react-router-dom";


it('should render correctly', () => {
	const tree = renderer
		.create(<BrowserRouter><Content/></BrowserRouter>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
