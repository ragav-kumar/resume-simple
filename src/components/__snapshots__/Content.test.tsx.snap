// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="sc-iJuVqt boTDdZ"
>
  <div
    className="sc-bdfBQB ivtnaj"
  >
    <ul
      className="sc-dlfnuX"
    />
  </div>
</div>
`;
